src/backend/aspen/api/views/auth.py

```diff
@@ -1,38 +1,53 @@
 import os
+from functools import lru_cache
 from urllib.parse import urlencode
 
 from authlib.integrations.base_client.errors import OAuthError
-from authlib.integrations.starlette_client import OAuth
-from fastapi import APIRouter
+from authlib.integrations.starlette_client import OAuth, StarletteOAuth2App
+from fastapi import APIRouter, Depends
 from fastapi.responses import RedirectResponse
 from starlette.requests import Request
 
 import aspen.api.error.http_exceptions as ex
-from aspen.api.config.config import settings
+from aspen.api.config.config import get_settings, Settings
 
 # From the example here:
 # https://github.com/authlib/demo-oauth-client/tree/master/fastapi-google-login
 router = APIRouter()
 
-oauth = OAuth()
-auth0 = oauth.register(
-    "auth0",
-    client_id=settings.AUTH0_CLIENT_ID,
-    client_secret=settings.AUTH0_CLIENT_SECRET,
-    api_base_url=settings.AUTH0_BASE_URL,
-    access_token_url=settings.AUTH0_ACCESS_TOKEN_URL,
-    authorize_url=settings.AUTH0_AUTHORIZE_URL,
-    client_kwargs=settings.AUTH0_CLIENT_KWARGS,
-)
+
+@lru_cache()
+def get_auth0_client() -> StarletteOAuth2App:
+    # TODO - settings is an unhashable type, so we can't make it a Dependency
+    # *and* use the lru_cache decorator on this getter. We should probably use
+    # fastapi's built-in dependency caching instead of lru_cache() here, but
+    # it needs a little more thought first.
+    # see: https://github.com/tiangolo/fastapi/issues/1985
```

Comment:
thanks for the comment! could we instantiate a `Settings` object in the scope directly above this function (or at the app entry point and attach it to a global object, like in flask?), then pass in a hashable object containing the needed settings?

Reply:
yep, that's definitely one option! I think this section of the code (thus the giant comment) needs a little more work for sure, but I haven't figured out which approach would be best just yet.